include/circt/Dialect/RTL/Structure.td

```diff
@@ -118,20 +119,32 @@ def RTLInstanceOp : RTLOp<"instance", []> {
   let arguments = (ins StrAttr:$instanceName,
                        FlatSymbolRefAttr:$moduleName,
                        Variadic<AnyType>:$inputs);
-  let results = (outs);
+  let results = (outs Variadic<AnyType>);
+
+  let extraClassDeclaration = [{   
+    StringAttr getResultName(size_t i);
+  }];
   
   let assemblyFormat = [{
-     $instanceName $moduleName `(` $inputs `)` attr-dict `:` type($inputs)
+    $instanceName $moduleName `(` $inputs `)` custom<ResultNames>( attr-dict ) `:` functional-type($inputs, results)
```

Comment:
Ooh fancy use of "custom"!

Reply:
It's [not in mainline yet](https://llvm.discourse.group/t/declarative-asm-format-custom-directive-on-attr-dict/2048). This is one of the advantages of having our own fork -- we can play around and suggest things.

Reply:
ah ok, be careful with this.  If we drift from mainline and carry our own patches, that can quickly turn into technical debt.

Reply:
100% agree